subsys/retention/retention.c

```diff
@@ -96,7 +96,7 @@ static int retention_checksum(const struct device *dev, uint32_t *output)
 		*output = 0;
 
 		while (pos < end) {
-			uint8_t read_size = MIN((end - pos), sizeof(buffer));
+			uint16_t read_size = MIN((end - pos), sizeof(buffer));
```

Comment:
Don't you want to just make it uint32_t? This is probably going to take full stack line, on 32-bit devices, anyway.
You may as well use unsigned int here.

Reply:
I am just asking because on some archs you may end up taking CPU word anyway, just there will be some and or other op used to take the lower 16bits.